packet/checksum.go

```diff
@@ -118,11 +120,16 @@ func SetPseudoHdrChecksum(p *Packet) {
 	ipv4, ipv6, _ := p.ParseAllKnownL3()
 	if ipv4 != nil {
 		p.GetIPv4().HdrChecksum = 0
-		tcp, udp, _ := p.ParseAllKnownL4ForIPv4()
+		tcp, udp, icmp := p.ParseAllKnownL4ForIPv4()
 		if tcp != nil {
 			p.GetTCPForIPv4().Cksum = SwapBytesUint16(CalculatePseudoHdrIPv4TCPCksum(p.GetIPv4()))
 		} else if udp != nil {
 			p.GetUDPForIPv4().DgramCksum = SwapBytesUint16(CalculatePseudoHdrIPv4UDPCksum(p.GetIPv4(), p.GetUDPForIPv4()))
+		} else if icmp != nil {
+			// calculate full software checksum for icmp, cause there is no
```

Comment:
Function SetPseudoHdrChecksum is not expected to calculate full checksums... @gshimansky what do you think?
I think it may be ok, but at least this should be stressed in func description

Reply:
I agree, but can't offer better variant. It is convenient to call SetPseudoHdrChecksum for packet and do nothing..there is no hard cksum for icmp and if we have possible tcp, udp, icmp packets with hwol mode, we need to parse packet ourselves, check whether it is icmp, calculate software cksum.

Reply:
We can rename the function from `SetPseudoHdrChecksum` to something like `SetHWOffloadingHdrChecksum`, so it would be clear that this function is used when HW offloading is enabled. In case of ICMP this mode still requires software calculation, but for this we have a comprehensive comment here.